libs/net/nodenetworkconfigurationpolicy.py

```diff
@@ -0,0 +1,167 @@
+
+
+@dataclass
+class BridgeOptions:
+    stp: STP
+
+
+@dataclass
+class Port:
+    name: str
+
+
+@dataclass
+class Bridge:
+    options: BridgeOptions | None = None
+    port: list[Port] | None = None
+    allow_extra_patch_ports: bool | None = None
+
+
+@dataclass
+class IP:
+    dhcp: bool
+    auto_dns: bool = True
+    enabled: bool = False
+
+
+@dataclass
+class IPv4(IP):
+    pass
+
+
+@dataclass
+class IPv6(IP):
+    autoconf: bool = False
+
+
+@dataclass
+class Interface:
+    name: str
+    type: str
+    state: str
+    ipv4: IPv4
+    ipv6: IPv6
+    bridge: Bridge | None = None
+
+
+@dataclass
+class DesiredState:
+    interfaces: list[Interface]
+
+
+class NNCPSuccessStatusNotMet(Exception):
+    pass
+
+
+class NetLibsNodeNetworkConfigurationPolicy(NodeNetworkConfigurationPolicy):
+    """
+    NodeNetworkConfigurationPolicy object.
+    """
+
+    api_group = Resource.ApiGroup.NMSTATE_IO
+
+    class Conditions:
+        class Reason:
+            SUCCESSFULLY_CONFIGURED = "SuccessfullyConfigured"
+            FAILED_TO_CONFIGURE = "FailedToConfigure"
+            NO_MATCHING_NODE = "NoMatchingNode"
+
+    def __init__(
+        self,
+        name: str,
+        desired_state: DesiredState,
+        node_selector: dict[str, str] | None = None,
+    ):
+        """
+        Create and manage NodeNetworkConfigurationPolicy
+
+        Args:
+            name (str): Name of the NodeNetworkConfigurationPolicy object.
+            desired_state (DesiredState): Desired policy configuration - interface creation, modification or removal.
+            client (DynamicClient, optional): Optional DynamicClient to use.
+            node_selector (dict, optional): A node selector that specifies the nodes to apply the node network
+                configuration policy to.
+        """
+        super().__init__(
+            name=name,
+        )
+        self._desired_state = desired_state
+        self._node_selector = node_selector
+
+    @retry(
+        wait_timeout=WAIT_FOR_STATUS_TIMEOUT, sleep=WAIT_FOR_STATUS_SLEEP, exceptions_dict={NNCPSuccessStatusNotMet: []}
+    )
+    def wait_for_status_success(self) -> list[dict]:
+        for condition in self.instance.status.conditions:
+            if condition["status"] == self.Condition.Status.TRUE:
+                if condition["reason"] == self.Conditions.Reason.SUCCESSFULLY_CONFIGURED:
+                    self.logger.info(f"{self.kind} {self.name} configured successfully")
+                    return condition
+
+                if (
+                    condition["reason"] == self.Conditions.Reason.NO_MATCHING_NODE
+                    or condition["reason"] == self.Conditions.Reason.FAILED_TO_CONFIGURE
+                ):
+                    raise NNCPConfigurationFailed(
+                        f"{self.kind} {self.name} failed to configure on condition: {condition}\n"
+                    )
+
+        raise NNCPSuccessStatusNotMet(f"{self.name}. None of the conditions were met.")
```

Comment:
Suggestion:
Refactored to reduce nesting by filtering conditions upfront, improving readability and maintainability. This makes the logic clearer and avoids redundant checks.

```suggestion
    def wait_for_status_success(self) -> dict:
        conditions = [
            condition
            for condition in self.instance.status.conditions
            if condition["status"] == self.Condition.Status.TRUE
        ]
    
        for condition in conditions:
            if condition["reason"] == self.Conditions.Reason.SUCCESSFULLY_CONFIGURED:
                self.logger.info(f"{self.kind} {self.name} configured successfully")
                return condition
    
            if condition["reason"] in {
                self.Conditions.Reason.NO_MATCHING_NODE,
                self.Conditions.Reason.FAILED_TO_CONFIGURE,
            }:
                raise NNCPConfigurationFailed(
                    f"{self.kind} {self.name} failed to configure on condition: {condition}\n"
                )
    
        raise NNCPSuccessStatusNotMet(f"{self.name}. None of the conditions were met.")
```